app/controllers/saml_idp_controller.rb

```diff
@@ -15,6 +16,20 @@ class SamlIdpController < ApplicationController
   before_action :store_sp_data, only: :auth
   before_action :confirm_two_factor_authenticated, except: [:metadata, :logout]
 
+  helper_method :saml_response_url
+
+  def raw_algorithm
+    SamlIdp.config.algorithm
+  end
+
+  def signature_opts
+    algorithm
+  end
+
+  def saml_response_url
+    saml_request.response_url
+  end
```

Comment:
I have already added the `response_url` method to the `SamlIdp::Request` class as part of my upstream PR. This is just a small convenience wrapper in our app controller. 

I added the `helper_method` because the 18F fork of the saml_idp controller class had that -- I did not verify if it was used in this app.

My preference would be to do away with the convenience method altogether since (a) it is not actually used as a helper, and (b) one more level of indirection is not a big savings compared to a few extra keystrokes.